### DISCUSSION TITLE
Is RangeDelete provide the transactional assurance？

### DISCUSSION THREAD

**Discussion by lsytj0413** [oxia-db/oxia#397] [Q&A] [answered]
As far as i understand, `DeleteRange` is dispatched to all shard, and there may some shard success, and other failed？

**Comment by merlimat** [comment, marked as answer]
There is no multi-key transactional guarantee, including `RangeDelete`, `List`. This was a design decision to allow for sharding, without implementing a complex transaction protocol on top of it.

**Reply by lsytj0413** [reply]
Ok，should we document this on `DeleteRange` api？and how can we use List to implement `Snapshot Read`？

**Reply by merlimat** [reply]
Yes, definitely

**Comment by merlimat** [comment]
@lsytj0413  To provide an update on this question. Since the introduction of the `PartitionKey()` option (see https://github.com/streamnative/oxia/releases/tag/v0.5.0), it's now possible to have an atomic `DeleteRange()` operation. 

If the partition-key is passed, the delete-range will be applied atomically on 1 single shard.
